growth/too/models.py

```diff
@@ -945,6 +947,164 @@ class Observation(db.Model):
+            if observation.field.ipix is not None
+            for i in observation.field.ipix}
+
+    def area(self, filt, start_time, end_time):
+        nside = Localization.nside
+        ipix = self.ipix(filt, start_time, end_time)
+        return hp.nside2pixarea(nside, degrees=True) * len(ipix)
+
+    def nexp(self, filt, start_time, end_time):
+        observations = self.get_observations(filt,
+                                             start_time, end_time)
+        observation_ids = []
+        for ii, observation in enumerate(observations):
+            if observation.observation_id not in observation_ids:
+                observation_ids.append(observation.observation_id)
+        return len(observation_ids)
+
+    def starttime(self, filt, start_time, end_time):
+        observations = self.get_observations(filt,
+                                             start_time, end_time)
+        if len(observations) == 0:
+            return -1
+        cnt = 0
+        for ii, observation in enumerate(observations):
+            if cnt == 0:
+                tt = observation.obstime
+            else:
+                tt = min(observation.obstime, tt)
+            cnt = cnt + 1
+        return tt
+
+    def totaltime(self, filt, start_time, end_time):
+        observations = self.get_observations(filt,
+                                             start_time, end_time)
+        tot = 0
+        observation_ids = []
+        for ii, observation in enumerate(observations):
+            if observation.observation_id not in observation_ids:
+                tot = tot + observation.exposure_time/60.0
+                observation_ids.append(observation.observation_id)
+        return tot
+
+    def limmag(self, filt, start_time, end_time):
+        observations = self.get_observations(filt,
+                                             start_time, end_time)
+        limmags = []
+        for ii, observation in enumerate(observations):
+            if observation.limmag is not None:
+                limmags.append(observation.limmag)
+        return np.median(limmags)
+
+    def get_probability(self, filt, start_time, end_time):
+        localization = Localization.query.filter_by(
+            dateobs=self.dateobs,
+            localization_name=self.localization_name).one()
+
+        ipix = np.asarray(list(self.ipix(filt, start_time, end_time)))
+        if len(ipix) > 0:
+            return localization.flat_2d[ipix].sum()
+        else:
+            return 0.0
+
+    def get_observations(self, filt, start_time, end_time):
+
+        localization = Localization.query.filter_by(
+            dateobs=self.dateobs,
+            localization_name=self.localization_name).one()
+        prob = localization.flat_2d
+
+        fields = Field.query.filter_by(telescope=self.telescope).all()
+        field_prob_1, field_ids_1 = [], []
+        field_prob_2, field_ids_2 = [], []
+        for field in fields:
+            if field.field_id < 800:
```

Comment:
What is special about field ID 800?

Reply:
Field ID 800 was implemented to differentiate between primary and secondary grids for ZTF, but as it isn't relevant for DECam I've changed the code to only do this if the telescope is ZTF.